quantum/process_keycode/process_autocorrect.c

```diff
@@ -39,28 +46,36 @@ bool process_autocorrection(uint16_t keycode, keyrecord_t* record) {
         case KC_LSFT:
         case KC_RSFT:
             return true;
-#    ifndef NO_ACTION_TAPPING
+#ifndef NO_ACTION_TAPPING
         case QK_MOD_TAP ... QK_MOD_TAP_MAX:
             if (((keycode >> 8) & 0xF) == MOD_LSFT) {
                 return true;
             }
-#        ifndef NO_ACTION_LAYER
+#    ifndef NO_ACTION_LAYER
         case QK_LAYER_TAP ... QK_LAYER_TAP_MAX:
-#        endif
+#    endif
             if (record->event.pressed || !record->tap.count) {
                 return true;
             }
             keycode &= 0xFF;
             break;
-#    endif
-#    ifndef NO_ACTION_ONESHOT
+#endif
+#ifdef SWAP_HANDS_ENABLE
+        case QK_SWAP_HANDS ... QK_SWAP_HANDS_MAX:
+            if (keycode >= 0x56F0 || record->event.pressed || !record->tap.count) {
+                return true;
+            }
+            keycode &= 0xFF;
+            break;
+#endif
+#ifndef NO_ACTION_ONESHOT
         case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
             if ((keycode & 0xF) == MOD_LSFT) {
                 return true;
             }
```

Comment:
I'd add "// Fallthrough intended." here too. It's important to reach the logic in the default case, which disables autocorrect when a one-shot mods other than shift is active.